### PR TITLE
Add support for using PR Creation Time in the feature allocator 

### PR DESCRIPTION
**Pull request by Aayyush**
This PR adds support for using the Unix time to evaluate feature allocation. It will be used to handle PRs in-flight during feature rollouts. 

Note: This feature flag implementation does not understand timestamps in the rule definition. So, we use Unix() time in seconds to evaluate the feature flag.

### CODE REVIEW

File: server/events/command/vcs.go [hunk 1]
```diff
@@ -22,13 +22,14 @@ func (d *VCSStatusUpdater) UpdateCombined(ctx context.Context, repo models.Repo,
...
-		Description: descrip,
-		DetailsURL:  "",
+		Repo:           repo,
+		PullNum:        pull.Num,
+		Ref:            pull.HeadCommit,
+		StatusName:     src,
+		State:          status,
+		Description:    descrip,
+		DetailsURL:     "",
+		PrCreationTime: pull.CreatedAt,
```
Comment (nishkrishnan) [inline server/events/command/vcs.go:32]: Keep this consistent with `PullNum`. so `PullCreationTime`.

Ideally, we'd group these under one pull object though, but fine with leaving as is.

Comment (nishkrishnan) [reply server/events/command/vcs.go:32]: Same for all others btw.

File: server/lyft/command/feature_runner.go [hunk 2]
```diff
@@ -44,7 +46,8 @@ func (r *PlatformModeFeatureRunner) Run(ctx *command.Context, cmd *command.Comme
 		return
 	}
 
-	shouldAllocate, err := r.featureAllocator.ShouldAllocate(feature.PlatformMode, ctx.HeadRepo.FullName)
+	// prCreationTime defaults to time.Unix(0, 0) to ensure features are turned on by default
+	shouldAllocate, err := r.featureAllocator.ShouldAllocate(feature.PlatformMode, ctx.HeadRepo.FullName, time.Unix(0, 0))
```
Comment (nishkrishnan) [inline server/lyft/command/feature_runner.go:50]: this feels weird that the caller should know this each time. 

Consider a request object and have the featureAllocator itself only care about the default.

File: server/events/command/vcs.go (hunk 1 above)
Comment (Aayyush) [reply server/events/command/vcs.go]: Done

File: server/lyft/command/feature_runner.go (hunk 2 above)
Comment (Aayyush) [reply server/lyft/command/feature_runner.go]: I added a FeatureContext object which basically has all the metadata needed to determine if a feature is enabled or not. This way, we can also add other attributes in the future to determine the eligibility and not have to worry about passing in the defaults. Lmk what you think!

**Review by nishkrishnan — APPROVED** [review]
Bonus points if you can get rid of the pego mock shit for our feature allocator.

**Comment by Aayyush** [comment]
> Bonus points if you can get rid of the pego mock shit for our feature allocator.

I can do it in a follow up task. Created a jira ticket: https://jira.lyft.net/browse/WENGINES-4694

**Pull request merged**